packages/core/contracts/optimistic-assertor/implementation/OptimisticAssertor.sol

```diff
@@ -97,15 +97,17 @@ contract OptimisticAssertor is Lockable, OptimisticAssertorInterface, Ownable {
             expirationTime: block.timestamp + liveness
         });
 
-        // Check if the Sovereign Security Manager is configured to arbitrate via DVM. Note that this call will revert
-        // if the Sovereign Security Manager is configured to not allow this assertion, such as if the manager has a
-        // configured whitelist and the asserter is not on it.
-        assertions[assertionId].respectDvmOnArbitration = _checkIfShouldRespectDvmOnArbitrate(assertionId);
+        SovereignSecurityManagerInterface.AssertionPolicies memory assertionPolicies =
+            _getSovereignSecurityManager(assertionId).getAssertionPolicies(assertionId);
+
+        // Check if the assertion is allowed by the sovereign security manager.
+        require(assertionPolicies.allowAssertion, "Assertion not allowed");
+
+        // Check if the Sovereign Security Manager is configured to arbitrate via DVM
+        assertions[assertionId].respectDvmOnArbitration = assertionPolicies.useDisputeResolution;
```

Comment:
I think that the `assertions[assertionId].respectDvmOnArbitration` name is not very accurate. Maybe it should be `assertions[assertionId].respectOracleOnArbitration` as the dispute resolution result could be from the sovereign security manager as well.

Reply:
Better have the same property name on both sides. So either `useDisputeResolution` or  `respectOracleOnArbitration` both in `assertions[assertionId]` and `assertionPolicies`